pkg/providers/tinkerbell/reconciler/reconciler.go

```diff
@@ -95,8 +95,15 @@ func (r *Reconciler) Reconcile(ctx context.Context, log logr.Logger, cluster *an
 		return controller.Result{}, err
 	}
 
-	return controller.NewPhaseRunner[*Scope]().Register(
-		r.ValidateControlPlaneIP,
```

Comment:
Alternatively, you could add this logic inside `ipValidator.ValidateControlPlaneIP` and avoid having to modify all reconcilers
That's the advnatage of having the shared logic in a common place